### DISCUSSION TITLE
Alpinejs v3: Alpine Expression Error: func(...).catch is not a function

### DISCUSSION THREAD

**Discussion by jet10000** [alpinejs/alpine#2487] [1. Help] [answered]
![image](https://user-images.githubusercontent.com/2258120/145702845-6b54d8a9-3317-4896-85aa-38e75265c3a1.png)
![image](https://user-images.githubusercontent.com/2258120/145702850-8b21578e-bd83-4464-b102-00e3c35ccc56.png)
![image](https://user-images.githubusercontent.com/2258120/145702858-4fd7abd3-c18b-4bc5-8d41-a25f2993c2d6.png)

# ho

**Comment by jet10000** [comment]
vite.config.js

![image](https://user-images.githubusercontent.com/2258120/145702903-cefa9d6e-75da-4624-b45d-8897abcca471.png)

**Comment by danddanddand** [comment, marked as answer]
Change your target from es2015 to es2017

**Reply by iautom8things** [reply]
This needs to be marked as the answer.  Thank you! 🙏

**Reply by bluediu** [reply]
Thank you!

**Reply by gjm** [reply]
@danddanddand I'm having the same problem but I'm using webpack not vite. 

How can I do this using webpack?

**Reply by jgerigmeyer** [reply]
@gjm It depends what loader you've configured Webpack to use for your JS. I'm using [swc-loader](https://swc.rs/docs/usage/swc-loader), and I was able to fix this by adding `jsc: { target: 'es2017' }` to my [SWC config](https://swc.rs/docs/configuration/compilation#jsctarget). If using [babel-loader](https://github.com/babel/babel-loader), you can similarly configure targets to something that has proper async support.

**Reply by t-raven** [reply]
for anyone dealing with laravel mix - [this extension](https://laravel-mix.com/extensions/swc) helped
